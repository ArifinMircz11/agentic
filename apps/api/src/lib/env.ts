import 'dotenv/config'

import { z } from 'zod'

export const envSchema = z.object({
  NODE_ENV: z
    .enum(['development', 'test', 'production'])
    .default('development'),
  DATABASE_URL: z.string().url(),
  JWT_SECRET: z.string()
})

// eslint-disable-next-line no-process-env
export const env = envSchema.parse(process.env)

export const isProd = env.NODE_ENV === 'production'
