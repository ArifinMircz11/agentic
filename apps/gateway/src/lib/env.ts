import type { Simplify } from 'type-fest'
import { parseZodSchema } from '@agentic/platform-core'
import {
  envSchema as baseEnvSchema,
  parseEnv as parseBaseEnv
} from '@agentic/platform-hono'
import { z } from 'zod'

export const envSchema = baseEnvSchema
  .extend({
    AGENTIC_API_BASE_URL: z.string().url(),
    AGENTIC_API_KEY: z.string().nonempty(),

    DO_RATE_LIMITER: z.custom<DurableObjectNamespace>(
      (ns) => ns && typeof ns === 'object'
    )
  })
  .strip()
export type RawEnv = z.infer<typeof envSchema>
export type Env = Simplify<ReturnType<typeof parseEnv>>

export function parseEnv(inputEnv: Record<string, unknown>) {
  const baseEnv = parseBaseEnv({
    SERVICE: 'gateway',
    ...inputEnv
  })

  const env = parseZodSchema(
    envSchema,
    { ...inputEnv, ...baseEnv },
    {
      error: 'Invalid environment variables'
    }
  )

  return {
    ...baseEnv,
    ...env
  }
}
