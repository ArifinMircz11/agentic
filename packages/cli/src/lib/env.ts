export const agenticApiBaseUrl =
  process.env.AGENTIC_API_BASE_URL || 'https://api.agentic.so'
