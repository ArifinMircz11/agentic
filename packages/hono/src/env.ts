import 'dotenv/config'

import type { Simplify } from 'type-fest'
import { parseZodSchema } from '@agentic/platform-core'
import { z } from 'zod'

import { logLevelsSchema } from './logger'

export const envSchema = z
  .object({
    ENVIRONMENT: z
      .enum(['development', 'test', 'production'])
      .default('development'),

    SERVICE: z.enum(['api', 'gateway']),

    LOG_LEVEL: logLevelsSchema.default('info'),

    SENTRY_DSN: z.string().url().optional()
  })
  .strip()

export function parseEnv(inputEnv: unknown) {
  const env = parseZodSchema(envSchema, inputEnv, {
    error: 'Invalid environment variables'
  })

  const isDev = env.ENVIRONMENT === 'development'
  const isTest = env.ENVIRONMENT === 'test'
  const isProd = env.ENVIRONMENT === 'production'
  const isBrowser = (globalThis as any).window !== undefined

  if (isProd && !env.SENTRY_DSN) {
    throw new Error(
      'Internal error: missing required "SENTRY_DSN" environment variable in production'
    )
  }

  return {
    ...env,
    isDev,
    isTest,
    isProd,
    isBrowser
  }
}

export type Env = Simplify<ReturnType<typeof parseEnv>>
