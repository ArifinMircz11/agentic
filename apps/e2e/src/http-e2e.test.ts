import contentType from 'fast-content-type-parse'
import defaultKy from 'ky'
import { describe, expect, test } from 'vitest'

import { env } from './env'
import { fixtureSuites } from './http-fixtures'

const ky = defaultKy.extend({
  prefixUrl: env.AGENTIC_GATEWAY_BASE_URL,

  // Disable automatic retries for testing.
  retry: 0,

  // Some tests expect HTTP errors, so handle them manually instead of throwing.
  throwHttpErrors: false
})

for (const [i, fixtureSuite] of fixtureSuites.entries()) {
  const { title, fixtures, compareResponseBodies = false } = fixtureSuite

  const describeFn = fixtureSuite.only ? describe.only : describe
  describeFn(title, () => {
    let fixtureResponseBody: any | undefined

    for (const [j, fixture] of fixtures.entries()) {
      const method = fixture.request?.method ?? 'GET'
      const timeout = fixture.timeout ?? 30_000
      const {
        status = 200,
        contentType: expectedContentType = 'application/json',
        headers: expectedHeaders,
        body: expectedBody,
        validate
      } = fixture.response ?? {}
      const snapshot =
        fixture.response?.snapshot ??
        fixtureSuite.snapshot ??
        (status >= 200 && status < 300)
      const debugFixture = !!(
        fixture.debug ??
        fixtureSuite.debug ??
        fixture.only ??
        fixtureSuite.only
      )
      const fixtureName = `${i}.${j}: ${method} ${fixture.path}`

      let testFn = (fixture.only ?? fixture.debug) ? test.only : test
      if (fixtureSuite.sequential) {
        testFn = testFn.sequential
      }

      testFn(
        fixtureName,
        {
          timeout
        },
        // eslint-disable-next-line no-loop-func
        async () => {
          const res = await ky(fixture.path, {
            timeout,
            ...fixture.request
          })

          if (res.status !== status && res.status >= 500) {
            let body: any
            try {
              body = await res.json()
            } catch {}

            console.error(`${fixtureName} => UNEXPECTED ERROR ${res.status}`, {
              body
            })
          }

          expect(res.status).toBe(status)

          const { type } = contentType.safeParse(
            res.headers.get('content-type') ?? ''
          )
          expect(type).toBe(expectedContentType)

          let body: any

          if (type.includes('json')) {
            try {
              body = await res.json()
            } catch (err) {
              console.error('json error', err)
              throw err
            }
          } else if (type.includes('text')) {
            body = await res.text()
          } else {
            body = await res.arrayBuffer()
          }

          if (expectedBody) {
            expect(body).toEqual(expectedBody)
          }

          if (validate) {
            await Promise.resolve(validate(body))
          }

          if (snapshot) {
            expect(body).toMatchSnapshot()
          }

          if (expectedHeaders) {
            for (const [key, value] of Object.entries(expectedHeaders)) {
              expect(res.headers.get(key)).toBe(value)
            }
          }

          if (compareResponseBodies && status >= 200 && status < 300) {
            if (!fixtureResponseBody) {
              fixtureResponseBody = body
            } else {
              expect(body).toEqual(fixtureResponseBody)
            }
          }

          if (debugFixture) {
            console.log(`${fixtureName} => ${res.status}`, {
              body,
              headers: Object.fromEntries(res.headers.entries())
            })
          }
        }
      )
    }
  })
}
