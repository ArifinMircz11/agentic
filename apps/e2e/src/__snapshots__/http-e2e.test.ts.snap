// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HTTP => MCP origin basic "add" tool > 6.0: POST @dev/test-basic-mcp/add 1`] = `
[
  {
    "text": "42",
    "type": "text",
  },
]
`;

exports[`HTTP => MCP origin basic "add" tool > 6.1: GET @dev/test-basic-mcp/add 1`] = `
[
  {
    "text": "42",
    "type": "text",
  },
]
`;

exports[`HTTP => OpenAPI origin basic GET caching > 4.0: GET @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "aut dicta possimus sint mollitia voluptas commodi quo doloremque
iste corrupti reiciendis voluptatem eius rerum
sit cumque quod eligendi laborum minima
perferendis recusandae assumenda consectetur porro architecto ipsum ipsam",
  "id": 13,
  "title": "dolorum ut in voluptas mollitia et saepe quo animi",
  "userId": 2,
}
`;

exports[`HTTP => OpenAPI origin basic GET caching > 4.1: GET @dev/test-basic-openapi/getPost?postId=13 1`] = `
{
  "body": "aut dicta possimus sint mollitia voluptas commodi quo doloremque
iste corrupti reiciendis voluptatem eius rerum
sit cumque quod eligendi laborum minima
perferendis recusandae assumenda consectetur porro architecto ipsum ipsam",
  "id": 13,
  "title": "dolorum ut in voluptas mollitia et saepe quo animi",
  "userId": 2,
}
`;

exports[`HTTP => OpenAPI origin basic GET caching > 4.2: GET @dev/test-basic-openapi/get_post?postId=13 1`] = `
{
  "body": "aut dicta possimus sint mollitia voluptas commodi quo doloremque
iste corrupti reiciendis voluptatem eius rerum
sit cumque quod eligendi laborum minima
perferendis recusandae assumenda consectetur porro architecto ipsum ipsam",
  "id": 13,
  "title": "dolorum ut in voluptas mollitia et saepe quo animi",
  "userId": 2,
}
`;

exports[`HTTP => OpenAPI origin basic POST caching > 5.0: POST @dev/test-basic-openapi/get_post 1`] = `
{
  "body": "aut dicta possimus sint mollitia voluptas commodi quo doloremque
iste corrupti reiciendis voluptatem eius rerum
sit cumque quod eligendi laborum minima
perferendis recusandae assumenda consectetur porro architecto ipsum ipsam",
  "id": 13,
  "title": "dolorum ut in voluptas mollitia et saepe quo animi",
  "userId": 2,
}
`;

exports[`HTTP => OpenAPI origin basic POST caching > 5.1: POST @dev/test-basic-openapi/get_post 1`] = `
{
  "body": "aut dicta possimus sint mollitia voluptas commodi quo doloremque
iste corrupti reiciendis voluptatem eius rerum
sit cumque quod eligendi laborum minima
perferendis recusandae assumenda consectetur porro architecto ipsum ipsam",
  "id": 13,
  "title": "dolorum ut in voluptas mollitia et saepe quo animi",
  "userId": 2,
}
`;

exports[`HTTP => OpenAPI origin basic bypass caching > 3.0: GET @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
  "id": 9,
  "title": "nesciunt iure omnis dolorem tempora et accusantium",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic bypass caching > 3.1: GET @dev/test-basic-openapi/getPost?postId=9 1`] = `
{
  "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
  "id": 9,
  "title": "nesciunt iure omnis dolorem tempora et accusantium",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic bypass caching > 3.2: GET @dev/test-basic-openapi/get_post?postId=9 1`] = `
{
  "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
  "id": 9,
  "title": "nesciunt iure omnis dolorem tempora et accusantium",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic bypass caching > 3.3: GET @dev/test-basic-openapi/get_post?postId=9 1`] = `
{
  "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
  "id": 9,
  "title": "nesciunt iure omnis dolorem tempora et accusantium",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic bypass caching > 3.4: GET @dev/test-basic-openapi/get_post?postId=9 1`] = `
{
  "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
  "id": 9,
  "title": "nesciunt iure omnis dolorem tempora et accusantium",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost errors > 1.8: POST @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost errors > 1.9: GET @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost success > 0.0: POST @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost success > 0.1: POST @dev/test-basic-openapi@latest/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost success > 0.2: GET @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost success > 0.3: GET @dev/test-basic-openapi/getPost?postId=1 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost success > 0.4: GET @dev/test-basic-openapi/get_post?postId=1 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin basic getPost success > 0.5: GET @dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`HTTP => OpenAPI origin default bypass caching > 2.0: GET @dev/test-everything-openapi/echo 1`] = `
{
  "postId": "9",
}
`;

exports[`HTTP => OpenAPI origin everything "echo" tool with empty body > 10.0: POST @dev/test-everything-openapi/echo 1`] = `{}`;

exports[`HTTP => OpenAPI origin everything "echo" tool with empty body > 10.1: POST @dev/test-everything-openapi/echo 1`] = `{}`;

exports[`HTTP => OpenAPI origin everything "pure" tool > 8.0: POST @dev/test-everything-openapi/pure 1`] = `
{
  "foo": "bar",
  "nala": "kitten",
}
`;

exports[`HTTP => OpenAPI origin everything "pure" tool > 8.1: POST @dev/test-everything-openapi/pure 1`] = `
{
  "foo": "bar",
  "nala": "kitten",
}
`;
